### ISSUE TITLE
v8.11.0 Unable to correctly use boolean.html property editor when creating toggle with custom plugin

### ISSUE THREAD

**Issue by EneldoDev** [umbraco/Umbraco-CMS#9910]
I have custom property editor plugin that allows me to create multiple different controls configurations. I've found that some configuration values are stored differently in database for default Umbraco property editors (in that case Toggle) and custom properties.

In this bug report I'll simplify the problem (without getting into details of my plugin), just to show what exactly I mean.

## Umbraco version

I am seeing this issue on Umbraco version: 8.11.0

Reproduction
------------

### Steps to reproduce

1. Create simple custom property editor with package.manifest as below (in my case it's much more complex, but it'll give the view on this issue):
```
{
  "propertyEditors": [
    {
      "name": "Custom toggle editor",
      "alias": "CustomToggle",
      "editor": {
        "view": "boolean",
        "valueType": "JSON"
      },
      "prevalues": {
        "fields": [
          {
            "label": "Default value",
            "key": "default",
            "view": "boolean"
          },
          {
            "label": "Show labels",
            "key": "showLabels",
            "view": "boolean"
          },
          {
            "label": "Label on",
            "key": "labelOn",
            "view": "textstring"
          },
          {
            "label": "Label off",
            "key": "labelOff",
            "view": "textstring"
          }
        ]
      }
    }
  ],
  "css": [
  ],
  "javascript": [
  ]
}
```
2. Using above property editor create Data type with below configuration:

- Default value - false;
- Show labels - true;
- Label on/ Label off: custom strings (In my example I'll use "Label on", and "Label off" values).

3. Create document type using above custom toggle control.
4. Create node with document type specified in point 3.

### Expected result

I'll see document type with editable toggle control and labels for toggle on and off options.

### Actual result
I see only toggle button without additional labels.

### Additional notes
In above example I've created copy of toggle control, but using custom property editor. It's configurtation in database was saved as below:
`{"default":"0","showLabels":"1","labelOn":"Label on","labelOff":"Label off"}`
See that boolean values are stored as 0/1.
When I create toggle with build in property editor (Umbraco.TrueFalse) exactly the same configuration is stored as:
`{"default":false,"showLabels":true,"labelOn":"Label on","labelOff":"Label off"}`
So far nothing to bother as true/false and 1/0 should generate the same behavior, but once the view is generated with Umbraco\Views\propertyeditors\boolean\boolean.html, my toggle button has show-labels configuration:
**show-labels="1"**  (for build in toggle it's rendered as show-labels="true")
and condition to show labels is:
` ngIf: labelPosition === 'right' && **showLabels === 'true'** `

In this case there is no possibility to show toggle labels depending on value of customly creaded toggle button.

I've checked in v7.x version of Umbraco, and boolean.html had slight different definition for show-labels property.
V7x:
`show-labels="{{model.config.labelOn ? 'true': 'false'}}"`
v8.11:
`show-labels="{{model.config.showLabels}}"`

Extending v8 solution with condition that was implemented in v7x would make boolean.html view easier to use in custom plugins.

**Comment by nul800sebastiaan** [comment]
Thanks @EneldoDev for the explanation. I know that @bjarnef has changed this in PR https://github.com/umbraco/Umbraco-CMS/pull/8477 for v8.8.0, hopefully he'll have some ideas on whether this can be improved!

**Comment by bjarnef** [comment]
Not sure where you are expecting the labels to be shown next to the toggle? Are we talking about the rendered toggle in the property editor, which in you example just use default configuration of `boolean` property editor 

![image](https://user-images.githubusercontent.com/2919859/112026935-fdf32f80-8b36-11eb-8d4f-803a1d8bff88.png)

or are we talking about the two prevalue toggles which use the `boolean` prevalue editor view?

![image](https://user-images.githubusercontent.com/2919859/112026970-0a778800-8b37-11eb-89fb-0cc85e73264f.png)

**Comment by bjarnef** [comment]
Okay, I think the issue is the value is being changed from a boolean to a string when extending the default config:

![image](https://user-images.githubusercontent.com/2919859/112027915-f54f2900-8b37-11eb-8dac-ddb72df42d0a.png)

**Comment by EneldoDev** [comment]
Yes, exactly. I've just wanted to describe this, but you were faster :) 
If this condition:
`show-labels="{{model.config.showLabels}}" `
would be changed to 
`show-labels="{{model.config.showLabels ? 'true': 'false'}}"` then Label on/Label off would be visible no matter how boolean value for custom property editors is stored in the database.

**Comment by bjarnef** [comment]
This PR should fix the issue: https://github.com/umbraco/Umbraco-CMS/pull/10035

**Comment by nathanwoulfe** [comment]
Fixed in #10035 🎉

**Comment by nul800sebastiaan** [comment]
Great! I moved the labels to the PR!
